packages/dev/core/src/Physics/v2/physicsBody.ts

```diff
@@ -0,0 +1,170 @@
+import type { IPhysicsEnginePluginV2, MassProperties } from "./IPhysicsEnginePluginV2";
+import type { PhysicsShape } from "./physicsShape";
+import { Vector3 } from "../../Maths/math.vector";
+import type { Scene } from "../../scene";
+
+/**
+ *
+ */
+export class PhysicsBody {
+    /** @internal */
+    public _pluginData: any = undefined;
+
+    private _physicsPlugin: IPhysicsEnginePluginV2 | undefined;
+
+    /**
+     *
+     * @param scene
+     * @returns
+     */
+    constructor(scene: Scene) {
+        if (!scene) {
+            return;
+        }
+        this._physicsPlugin = scene.getPhysicsEngine()?.getPhysicsPlugin() as any;

```

Comment:
if this is undefined the physics body will not work at all. It should throw if there is no physics plugin available. This way the physics plugin doesn't have to be optional and save all of the `?` added

Reply:
Fixed